internal/security/vaultclient/vault.go

```diff
@@ -0,0 +1,300 @@
+	initResp := InitResponse{}
+	//rawBytes, err := ioutil.ReadFile(filepath.Join(config.TokenFolderPath, config.TokenFile))
+	readCloser := fileioperformer.MakeReadCloser(vmkReader)
+	rawBytes, err := ioutil.ReadAll(readCloser)
+	defer readCloser.Close()
+	if err != nil {
+		vc.logger.Error(fmt.Sprintf("failed to read the Vault JSON response init file: %s", err.Error()))
+		return 0, err
+	}
+
+	if err = json.Unmarshal(rawBytes, &initResp); err != nil {
+		vc.logger.Error(fmt.Sprintf("failed to build the JSON structure from the init response body: %s", err.Error()))
+		return 0, err
+	}
+
+	url := &url.URL{
+		Scheme: vc.scheme,
+		Host:   vc.host,
+		Path:   VaultUnsealAPI,
+	}
+
+	keyCounter := 1
+	for _, key := range initResp.KeysBase64 {
+		unsealRequest := UnsealRequest{
+			Key: key,
+		}
+		jsonBlock, err := json.Marshal(&unsealRequest)
+		if err != nil {
+			vc.logger.Error(fmt.Sprintf("failed to build the Vault unseal request (key shares parameter): %s", err.Error()))
+			return 0, err
+		}
+		req, err := http.NewRequest(http.MethodPost, url.String(), bytes.NewBuffer(jsonBlock))
+		req.Header.Set("Content-Type", JsonContentType)
+		resp, err := vc.client.Do(req)
+		if err != nil {
+			vc.logger.Error(fmt.Sprintf("failed to send the Vault init request: %s", err.Error()))
+			return 0, err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			vc.logger.Error(fmt.Sprintf("vault unseal request failed with status code: %s", resp.Status))
+			return resp.StatusCode, err
+		}
+
+		unsealedBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			vc.logger.Error(fmt.Sprintf("failed to fetch the Vault unseal request response body: %s", err.Error()))
+			return 0, err
+		}
+		unsealResponse := UnsealResponse{}
+		if err = json.Unmarshal(unsealedBody, &unsealResponse); err != nil {
+			vc.logger.Error(fmt.Sprintf("failed to build the JSON structure from the unseal request response body: %s", err.Error()))
+			return 0, err
+		}
+
+		vc.logger.Info(fmt.Sprintf("Vault key share %d/%d successfully applied.", keyCounter, config.VaultSecretShares))
+		if !unsealResponse.Sealed {
+			vc.logger.Info("Vault key share threshold reached. Unsealing complete.")
+			return resp.StatusCode, nil
+		}
+		keyCounter++
+	}
+	return 0, fmt.Errorf("%d", 1)
+}
+
+func (vc *vaultClient) InstallPolicy(
+	token authtokenreader.AuthTokenReader,
+	policyName string,
+	policyDocument string) (statusCode int, err error) {
+
+	path := fmt.Sprintf(CreatePolicyPath, url.PathEscape(policyName))
+	url := &url.URL{
+		Scheme: vc.scheme,
+		Host:   vc.host,
+		Path:   path,
+	}
+
+	request := UpdateAclPolicyRequest{Policy: policyDocument}
+	requestBytes, err := json.Marshal(request)
+	if err != nil {
+		vc.logger.Error(fmt.Sprintf("failed to build the Vault create policy request: %s", err.Error()))
+		return 0, err
+	}
+
+	req, err := http.NewRequest(http.MethodPut, url.String(), bytes.NewReader(requestBytes))
```

Comment:
Please handle this error `err` from `http.NewRequest`: either log it or ignore it.

Reply:
Not my code, but fixed it anyway.